cider/tests/utils/DuckDbQueryRunner.cpp

```diff
@@ -388,6 +395,47 @@ void addColumnDataToCiderBatch<CiderByteArray>(
   }
 }
 
+CiderBatch DuckDbResultConvertor::fetchOneArrowFormattedBatch(
+    std::unique_ptr<duckdb::DataChunk>& chunk,
+    std::vector<std::string>& names,
+    std::string& config_timezone) {
+  int col_num = chunk->ColumnCount();
+  int row_num = chunk->size();
+
+  // Construct ArrowSchema using methods from duckdb
+  std::vector<::duckdb::LogicalType> types = chunk->GetTypes();
+  auto arrow_schema = CiderBatchUtils::allocateArrowSchema();
+  ::duckdb::QueryResult::ToArrowSchema(arrow_schema, types, names, config_timezone);
+
+  // Construct ArrowArray
+  auto arrow_array = CiderBatchUtils::allocateArrowArray();
+  chunk->ToArrowArray(arrow_array);
+
+  // Build CiderBatch
+  auto batch = StructBatch::Create(
+      arrow_schema, std::make_shared<CiderDefaultAllocator>(), arrow_array);
+
+  // ToArrowArray() will not compute null_count, so we do it manually here
+  for (int i = 0; i < col_num; ++i) {
+    auto child = batch->getChildAt(i);
```

Comment:
make L420-433 as a method?

Reply:
It is now encapsulated into a private member function `updateChildrenNullCounts`. But I'm not sure if this name is proper.